spinnman/processes/get_machine_process.py

```diff
@@ -171,52 +185,259 @@ def get_machine_details(self, boot_x, boot_y, width, height,
-            logger.warning(msg)
-
-            if ignore_bad_ethernets:
-                # No chips actually removed
-                removed_chips = []
+                    logger.warning(
+                        "Not using chip {}:{} as it has an unexpected "
+                        "ip address: {}", chip_info.x, chip_info.y,
+                        chip_info.ethernet_ip_address)
+                    continue
+            else:
+                machine.add_chip(self._make_chip(chip_info, machine))
+
         machine.validate()
-        return machine_repair(machine, repair_machine, removed_chips)
+        return machine_repair(machine, repair_machine)
 
-    def get_chip_info(self):
-        """ Get the chip information for the machine.
+    # Stuff below here is purely for dealing with ignores
 
-        .. note::
-            :py:meth:`get_machine_details` must have been called first.
+    def _process_ignore_links(self, machine):
         """
-        return self._chip_info
+        Processes the collection of ignore links to remove then from chipinfo
+
+        Converts any local x, y ipaddress to global xy
+
+        Discards any ignores which are known to have no\
+        affect based on the already read chip_info
+
+        Also removes any inverse linka
+
+        Logs all actions except for ignores with unused ip addresses
+
+        :param machine: An empty machine to handle wraparrounds
+        """
+        if len(self._ignore_links) == 0:
+            return
+
+        discarded = set()
+        for ignore in self._ignore_links:
+            if len(ignore) == 3:
+                local_x, local_y, link = ignore
+                ip = None
+            else:
+                local_x, local_y, link, ip = ignore
+            global_xy = self._ignores_local_to_global(
+                local_x, local_y, ip, machine)
+            if global_xy is None:
+                continue
+            chip_info = self._chip_info[global_xy]
+            if link in chip_info.working_links:
+                chip_info.working_links.remove(link)
+                logger.info("On chip {} ignoring link:{}",
+                            global_xy, link)
+                # ignore the inverse link too
+                inv_xy = machine.xy_over_link(global_xy[0], global_xy[1], link)
+                discarded.add((global_xy, link))
+                if inv_xy in self._chip_info:
+                    inv_chip_info = self._chip_info[inv_xy]
+                    inv_link = (link + 3) % 6
+                    if inv_link in inv_chip_info.working_links:
+                        inv_chip_info.working_links.remove(inv_link)
+                        logger.info("On chip {} ignoring inverse link:{}",
+                                    inv_xy, inv_link)
+                        discarded.add((inv_xy, inv_link))
+                    # No log if the inverse does not exits
+            else:
+                if (global_xy, link) not in discarded:
```

Comment:
bad bad bad. if the user declared it to be ignored. they know more than the tools at this point. the user should always be right in this view

Reply:
No idea what you dislike here.

What this code is doing is saying if the user ignore the link in one direction automatically ignore in the other.

Is that a bad idea and if so why?

Reply:
ah. i misread the message. i read the message as :

"On chip {} link:{} found so discarding the ignore"

and that read as you were saying "we found a link, im adding it to the set, screw you ignore list"

Reply:
Improved th logr message